torch/autograd/gradcheck.py

```diff
@@ -718,46 +1040,26 @@ def fail_test(msg):
         return False
 
     func_out = func(*tupled_inputs)
-    outputs = _differentiable_outputs(func_out)
 
-    check_outputs(outputs)
+    if fast_mode and (any(is_tensor_like(o) and o.is_complex() for o in _as_tuple(func_out)) or
+                      any(is_tensor_like(i) and i.is_complex() for i in tupled_inputs)):
+        raise NotImplementedError("Fast mode for gradcheck and gradgradcheck is currently only implemented"
```

Comment:
Why not simply silently fall back to slow mode if anything is complex?
It would make subsequent PRs much simplers as you can always call with fast=True and it will automatically fallback to the slow version for complex tests.